libcxx/include/__algorithm/comp.h

```diff
@@ -11,6 +11,7 @@
 
 #include <__config>
```

Comment:
Please explain the optimization in the commit message. You can probably base it on the comment I drafted in another review comment.